Dockerfile

```diff
@@ -1,14 +1,16 @@
 FROM alpine:3.12
 
-RUN apk --no-cache add curl git
+RUN apk --update --no-cache add curl git
+
+# Use latest recommended version here
+ARG HELM_VERSION=3.3.0
+
+WORKDIR /usr/local/bin
 
 # Install Helm
-RUN curl -Lo helm-v3.1.2-linux-amd64.tar.gz https://get.helm.sh/helm-v3.1.2-linux-amd64.tar.gz && \
-    tar -zxvf helm-v3.1.2-linux-amd64.tar.gz && \
-    mv linux-amd64/helm /usr/local/bin/helm && \
-    rm helm-v3.1.2-linux-amd64.tar.gz && \
-    rm -rf linux-amd64 && \
-    helm version
+RUN curl -sSL https://get.helm.sh/helm-v${HELM_VERSION}-linux-amd64.tar.gz | tar xz \
+  --strip=1 linux-amd64/helm && \
```

Comment:
neat trick is strip removes the usual dance of extracting something into a dir then later removing the junk around it. This is especially easy when the tar includes a statically linked binary